### PR TITLE
Introducing `collect` `batch` `take` and `skip`

### PR DESCRIPTION
**Pull request by cloderic**

### CODE REVIEW

File: src/batch.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// @flow
+const { transduceToStream } = require('./transduce');
+
+import type { Stream } from './basics';
+import type { Transformer } from './transduce';
+
+function createBatchTransformer<T, AccumulationT>(count: number): Transformer<T, T[], AccumulationT> {
+  return (reducer) => {
+    let currentBatch = undefined;
```
Comment (alexisfontaine) [inline src/batch.js]: Why not use fixed size arrays?

File: src/batch.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    return (accumulation, event) => {
+      if (event.error) {
+        return reducer(accumulation, { error: event.error });
+      }
+
+      if (event.done) {
+        if (currentBatch != null) {
```
Comment (alexisfontaine) [inline src/batch.js]: Use strict inequality whenever possible.

File: src/batch.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          return reducer(reducerResult.accumulation, { done: true });
+        }
+        return reducer(accumulation, { done: true });
+      }
+
+      currentBatch = currentBatch || [];
+      currentBatch.push(event.value);
+      if (currentBatch.length >= count) {
+        const sentBatch = currentBatch;
+        currentBatch = undefined;
```
Comment (alexisfontaine) [inline src/batch.js]: Same as [here](https://github.com/craft-ai/alpes/pull/7/files#diff-76aa52b32899b0a26a582e6def699a55R9).

File: src/collect.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+function collect<T>(): (Stream<T>) => Promise<T[]> {
+  return transduce(
+    undefined,
+    (accumulation: T[], event: Event<T>) => {
+      if (event.error) {
+        throw event.error;
+      }
+
+      if (event.value != null) {
```
Comment (alexisfontaine) [inline src/collect.js]: Is there a good reason to skip `null` values here?
Do we expect `of(null).thru(collect())` to resolve to `[]`?

Comment (cloderic) [reply src/collect.js]: no it should resolve to `[null]`. It is fixed in ea0307a

File: src/batch.js (hunk 1 above)
Comment (cloderic) [reply src/batch.js]: RIght, done in 35d29c4

**Comment by cloderic** [comment]
Review taken into account @alexisfontaine

**Pull request merged**